ops/model.py

```diff
@@ -1094,20 +1098,80 @@ def _validate_content(cls, content: Optional[Dict[str, str]]):
 
     @property
     def id(self) -> Optional[str]:
-        """Unique identifier for this secret.
+        """Locator ID (URI) for this secret.
+
+        This has an unfortunate name for historical reasons, as it's not
+        really a unique identifier, but the secret's locator URI, which may or
+        may not include the model UUID (for cross-model secrets).
+
+        Charms should treat this as an opaque string for looking up secrets
+        and sharing them via relation data. If you need a charm-local "name"
+        for a secret, use a :attr:`label`. (If a charm needs a truly unique
+        identifier for identifying one secret in a set of secrets of arbitrary
+        size, use :attr:`unique_identifier` -- this should be rare.)
 
         This will be None if you obtained the secret using
         :meth:`Model.get_secret` with a label but no ID.
         """
         return self._id
 
+    @property
+    def unique_identifier(self) -> Optional[str]:
+        """Unique identifier of this secret.
+
+        This is the secret's globally-unique identifier (currently a
+        20-character Xid, for example "9m4e2mr0ui3e8a215n4g").
+
+        Most charms should use :attr:`id` (the secret's locator ID) to send
+        the secret's ID across relation data, and labels (:attr:`label`) to
+        assign a charm-local "name" to the secret for lookup in this charm.
+        However, ``unique_identifier`` can be useful to distinguish secrets in
+        cases where the charm has a set of secrets of arbitrary size, for
+        example, a group of 10 or 20 TLS certificates.
+
+        This will be None if you obtained the secret using
```

Comment:
Maybe mention it is always set if you are the owner

Reply:
This is not actually true: it will be `None` if you are the owner also (if you don't provide a label). This is how the API / modelling works: `get_secret` calls the `secret-get` hook tool only, but not `secret-info-get`, so it doesn't fetch the label.